overlord/configstate/configcore/kernel.go

```diff
@@ -192,6 +205,18 @@ func handleCmdlineAppend(c RunTransaction, opts *fsOnlyContext) error {
 	logger.Debugf("handling %v", kernelOpts)
 
 	st := c.State()
+	// If not seeded yet, this is coming from the gadget defaults and it is
+	// already applied to the kernel command line, do not create another
+	// change for this.
```

Comment:
I think a snap set system call during seeding results in a conflict? can you double check that?  in that case is worth mentioning that here, because in principle some snap with snapd-control could do their own "snap set" during seeding

Reply:
It looks like it is not the case, and no conflict is produced. I've changed this so now an error is returned if a set is tried while seeding, as it sounded risky to allow it. But let me know your thoughts.